<a href="https://colab.research.google.com/github/jatinkchaudhary/Classification-Problem-/blob/main/Parkinson's%20Disease%20Classifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Necessary Imports. 

In [ ]:
!pip install sklearn
!pip install --upgrade scikit-learn 
!pip install dataprep
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from sklearn.preprocessing import MinMaxScaler 
from xgboost import XGBClassifier 
import os, sys

Requirement already up-to-date: scikit-learn in /usr/local/lib/python3.7/dist-packages (0.24.2)


Reading the dataset. 

In [ ]:
df=pd.read_csv('parkinsons.data')
df.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...   spread2        D2       PPE
0  phon_R01_S01_1      119.992       157.302  ...  0.266482  2.301442  0.284654
1  phon_R01_S01_2      122.400       148.650  ...  0.335590  2.486855  0.368674
2  phon_R01_S01_3      116.682       131.111  ...  0.311173  2.342259  0.332634
3  phon_R01_S01_4      116.676       137.871  ...  0.334147  2.405554  0.368975
4  phon_R01_S01_5      116.014       141.781  ...  0.234513  2.332180  0.410335

[5 rows x 24 columns]

Exploratory Data Analysis 

Feature Extraction (all the columns except Status)

In [ ]:
features = df.loc[:, df.columns!='status'].values[:,1:]


Labels  (Status columns has the binary label for being infected with Parkinson's disease and not being infected.)

In [ ]:
labels= df.loc[:,'status'].values

Count of the Label

In [ ]:
print(labels[labels==1].shape[0])
print(labels[labels==0].shape[0])

147
48


This concludes that there are 147 positives and 48 negatives in the dataset present with us.

Normalization of the Data. 

In [ ]:
scalar=MinMaxScaler((-1,1))
#Scaling of the data and its parameters
x=scalar.fit_transform(features)
y=labels

Splitting the data into training and testing. 30% of the data is kept for testing purpose. 


In [ ]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.3, random_state=7)

eXtreme gradient boosting using XGBClassifier and training the model. 

---



In [ ]:
model=XGBClassifier()
#training the model 
model.fit(x_train, y_train)

XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

Check the model trained upon the test set and then calculate the accuracy of the model obtained. 

In [ ]:
y_pred= model.predict(x_test)
print(f'Accuracy Score=', accuracy_score(y_test, y_pred))
print(f'Accuracy Percentage=', (accuracy_score(y_test, y_pred)*100), '%')


Accuracy Score= 0.9152542372881356
Accuracy Percentage= 91.52542372881356 %
